### DISCUSSION TITLE
Proper tail calls - direct recursion: test fails

### DISCUSSION THREAD

**Discussion by ptcalex** [mozilla/rhino#1445] [General]
The table at https://compat-table.github.io/compat-table/es6/ marks direct recursion as "YES/Green" for Rhino 1.7.14, however if I run the test myself I get a java.lang.StackOverflowError.

Am I running the test wrong?

test.js
```
(function() {
  'use strict';
  return (function f(n) {
    if (n <= 0) { return "foo"; }
    return f(n-1);
  }(1e6)) === "foo";
})();
```
Running as

`
java -jar rhino-1.7.14.jar test.js 
`

and

`
java -jar rhino-1.7.14.jar -version 200 test.js 
`

**Comment by p-bakker** [comment]
While the compat table marks it as supported in Rhino, it's only supported when Rhino is run in interpreted mode.

Is it running in interpreted mode when you try to run the test?

**Reply by ptcalex** [reply]
Thank you Paul,
it makes sense, and in fact with `-opt -1` it works as intended.
